### ISSUE TITLE
[FE][NPE] StatisticExecutor NPE

### ISSUE THREAD

**Issue by kevincai** [StarRocks/starrocks#39837]
```
2024-01-24 08:18:46,975 WARN (AutoStatistic|40) [StatisticExecutor.collectStatistics():256] Collect statistics error
java.lang.NullPointerException: null
    at com.starrocks.statistic.FullStatisticsCollectJob.buildBatchCollectFullStatisticSQL(FullStatisticsCollectJob.java:243) ~[starrocks-fe.jar:?]
    at com.starrocks.statistic.FullStatisticsCollectJob.buildCollectSQLList(FullStatisticsCollectJob.java:227) ~[starrocks-fe.jar:?]
    at com.starrocks.statistic.FullStatisticsCollectJob.collect(FullStatisticsCollectJob.java:82) ~[starrocks-fe.jar:?]
    at com.starrocks.statistic.StatisticExecutor.collectStatistics(StatisticExecutor.java:254) ~[starrocks-fe.jar:?]
    at com.starrocks.statistic.StatisticAutoCollector.runAfterCatalogReady(StatisticAutoCollector.java:86) ~[starrocks-fe.jar:?]
    at com.starrocks.common.util.FrontendDaemon.runOneCycle(FrontendDaemon.java:72) ~[starrocks-fe.jar:?]
    at com.starrocks.common.util.Daemon.run(Daemon.java:115) ~[starrocks-fe.jar:?]
2024-01-24 08:18:46,975 WARN (AutoStatistic|40) [StatisticExecutor.collectStatistics():256] Collect statistics error
java.lang.NullPointerException: null
    at com.starrocks.statistic.FullStatisticsCollectJob.buildBatchCollectFullStatisticSQL(FullStatisticsCollectJob.java:243) ~[starrocks-fe.jar:?]
    at com.starrocks.statistic.FullStatisticsCollectJob.buildCollectSQLList(FullStatisticsCollectJob.java:227) ~[starrocks-fe.jar:?]
    at com.starrocks.statistic.FullStatisticsCollectJob.collect(FullStatisticsCollectJob.java:82) ~[starrocks-fe.jar:?]
    at com.starrocks.statistic.StatisticExecutor.collectStatistics(StatisticExecutor.java:254) ~[starrocks-fe.jar:?]
    at com.starrocks.statistic.StatisticAutoCollector.runAfterCatalogReady(StatisticAutoCollector.java:86) ~[starrocks-fe.jar:?]
    at com.starrocks.common.util.FrontendDaemon.runOneCycle(FrontendDaemon.java:72) ~[starrocks-fe.jar:?]
    at com.starrocks.common.util.Daemon.run(Daemon.java:115) ~[starrocks-fe.jar:?]
```

StarRocks version: 3.1.6

**Comment by dirtysalt** [comment]
SQL for reference.

> 2024-01-31 14:35:33,211 DEBUG (AutoStatistic|27) [FullStatisticsCollectJob.collectStatisticSync():115] statistics collect sql : SELECT cast(4 as INT), cast(583786 as BIGINT), 'k3', cast(COUNT(1) as BIGINT), cast(IFNULL(SUM(CHAR_LENGTH(`k3`)), 0) as BIGINT), hex(
hll_serialize(IFNULL(hll_raw(`k3`), hll_empty()))), cast(COUNT(1) - COUNT(`k3`) as BIGINT), IFNULL(MAX(LEFT(`k3`, 200)), ''), IFNULL(MIN(LEFT(`k3`, 200)), '')  FROM `test_broker_load_b06a2760_c002_11ee_bf69_00163e1bb865`.`duplicate_table_not_null` partition `dup
licate_table_not_null`

**Comment by ajayl83** [comment]
@kevincai can i pick this up?

**Comment by kevincai** [comment]
> @kevincai can i pick this up?

sure,  of course!

**Comment by ajayl83** [comment]
@kevincai is there exsting UT or IT to reproduce this issue?

**Comment by kevincai** [comment]
> @kevincai is there exsting UT or IT to reproduce this issue?

not sure if there is any. However, the root cause is clear. the partition is not locked during the execution of StatisticExecutor, it is possible that after the task done, the partition doesn't exist any more. The code should be tolerant of such case and check the existence before dereference the Partitionn for further processing.

**Comment by rohitrs1983** [comment]
@kevincai this issue is already fixed as part of https://github.com/StarRocks/starrocks/pull/40453

    protected List<List<String>> buildCollectSQLList(int parallelism) {
        List<String> totalQuerySQL = new ArrayList<>();
        for (Long partitionId : partitionIdList) {
            Partition partition = table.getPartition(partitionId);
            **if (partition == null) {
                // statistics job doesn't lock DB, partition may be dropped, skip it
                continue;
            }**
    }

**Comment by kevincai** [comment]
@rohitrs1983 thank you.

I didn't get a chance to double check if it is fixed or not in recent code.

Close this issue as fixed in #40453
